renderdoc/driver/vulkan/vk_memory.cpp

```diff
@@ -28,6 +28,66 @@
 RDOC_CONFIG(bool, Vulkan_Debug_MemoryAllocationLogging, false,
             "Output verbose debug logging messages when allocating internal memory.");
 
+void WrappedVulkan::TrackBufferAddress(VkDevice device, VkBuffer buffer)
+{
+  SCOPED_LOCK(m_DeviceAddressResourcesLock);
```

Comment:
As a minor improvement it would be better to reduce contention on this lock by just doing the check while holding the lock rather than all of the rest of the work, e.g:

```cpp
bool isBDA = false;
{
  SCOPED_LOCK(m_DeviceAddressResourcesLock);
  isBDA = m_DeviceAddressResources.IDs.contains(GetResID(buffer));
}
```